pkg/util/iterator/context.go

```diff
@@ -0,0 +1,7 @@
+package iterator
+
+import "context"
+
+var CustomIterator = func() ([]context.Context, error) {
+	return []context.Context{context.Background()}, nil
+}
```

Comment:
A couple of point here.

1. Can we make this not a global but something passed in config? 
2. why return a context ? It seems to be magic, should we return something like a tenant key or equivalent. Feels like this opacity is not helping anyone

Reply:
An example of this is around AccessControl. https://github.com/kumahq/kuma/blob/579c9626d4a3a087fbf3db2f5a7b3e329c935602/pkg/core/runtime/runtime.go#L83-L88 we can then set these in `Plugin.{Before,After}Bootstrap` with `ctx.WithAccess`

Reply:
So for "2." - we're leaving it as it is (so returning a context).